src/lints/inherent_method_now_doc_hidden.ron

```diff
@@ -0,0 +1,68 @@
+SemverQuery(
+    id: "inherent_method_now_doc_hidden",
+    human_readable_name: "inherent method #[doc(hidden)] added",
+    description: "A method or associated fn is now marked #[doc(hidden)] and is thus no longer part of the public API.",
+    required_update: Major,
+    lint_level: Deny,
+    reference_link: Some("https://doc.rust-lang.org/rustdoc/write-documentation/the-doc-attribute.html#hidden"),
+    query: r#"
+    {
+        CrateDiff {
+            baseline {
+                item {
+                    ... on ImplOwner {
+                        visibility_limit @filter(op: "=", value: ["$public"]) @output
+
+                        importable_path {
+                            path @output @tag
+                            public_api @filter(op: "=", value: ["$true"])
+                        }
+
+                        inherent_impl {
+                            method {
+                                method_visibility: visibility_limit @filter(op: "=", value: ["$public"]) @output
+                                method_name: name @output @tag
+                                public_api_eligible @filter(op: "=", value: ["$true"])
+
+                                span_: span @optional {
+                                    filename @output
+                                    begin_line @output
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+            current {
+                item {
+                    ... on ImplOwner {
+                        visibility_limit @filter(op: "=", value: ["$public"])
+                        name @output
+
+                        importable_path {
+                            path @filter(op: "=", value: ["%path"])
+                            public_api @filter(op: "=", value: ["$true"])
+                        }
+
+			# TODO - why different from const?
```

Comment:
What I mean is, I'm not sure if I have to worry about all of this:

https://github.com/obi1kenobi/cargo-semver-checks/blob/main/src/lints/inherent_method_const_removed.ron#L48

Reply:
No, you don't. This query looks correct as-is.

In the case of method removals, or removals of `const` on the method, the compiler may try to fall back to other definitions that match the name and/or const restriction as needed. Those other definitions may come from traits, which is why we need the logic you highlighted.

In this case, we don't care about that — we're looking for the exact same item, with the only change being that it's now `#[doc(hidden)]` and didn't use to be. The compiler doesn't avoid using `#[doc(hidden)]` items and wouldn't fall back to another definition of it, so this simpler logic is correct. The simplicity is why I recommended it as a starting lint :)